erpnext/public/js/utils.js

```diff
@@ -694,6 +694,97 @@ $(document).on('app_ready', function() {
 		});
 	}
 
+	$.each(["Purchase Receipt", "Stock Entry", "Delivery Note"], function(i, doctype) {
+		frappe.ui.form.on(doctype, {
+			refresh: function(frm) {
+				if(!frm.doc.is_return && frm.doc.status!="Closed") {
+					if (frm.doc.docstatus == 0) {
+						frm.add_custom_button(__('Quality Inspection'), function () {
+							let data = [];
+							let inspection_type_map = {
+								"Purchase Receipt": "Incoming",
+								"Delivery Note": "Outgoing",
+								"Stock Entry": "In Process"
+							}
+							let inspection_type = inspection_type_map[frm.doc.doctype];
+							data = frm.doc.items.map(row => ({
+								"doctype": row.doctype,
```

Comment:
Remove this line, it's causing the `read-only` to not apply. Only pass the data you want.